### DISCUSSION TITLE
how to remove local folders after add them to exclude sync_list

### DISCUSSION THREAD

**Discussion by elgarbe** [abraunegg/onedrive#2252] [Q&A] [answered]
Hi, I've copied several folders to my local onedrive folder in order to backup them. Now that they are on the cloud I add them to selective sync_list as excluded ones. I was hopping that local folders will be removed and remote files will be untouched. Is that the expected behavior, right? If not, then, how to safely delete local folders and preserve remote ones. 
Is remove_source_files options used for this? I can't find reference on the documentation for this parameter.

**Comment by abraunegg** [comment, marked as answer]
@elgarbe 
> I was hopping that local folders will be removed and remote files will be untouched. Is that the expected behavior, right?

Yes, this is the *correct* behavior. You found a bug, which has been fixed, so some *manual* cleanup done by you will need to be done.

This being said, in a normal situation (as the bug has now been fixed) those directories would not exist in the first place - so no issue.

Now - there is an option to 'clean up' local files - `--cleanup-local-files` .. however this is *only* for when using the `--download-only` command - so you are basically making an exact replica of the online state locally - including any remote deletes. This option is not viable for your use case - as that content exists online, and you are performing client side filtering via 'sync_list'

> If not, then, how to safely delete local folders and preserve remote ones.

Your only option is to manually delete the files | folders locally.

> Is remove_source_files options used for this? I can't find reference on the documentation for this parameter.

As per the man page and usage document:
```
--remove-source-files
      Remove source file after successful transfer to OneDrive when using --upload-only
```

So again, this is not applicable for your situation - as this is used for uploading to OneDrive (say a backup system or something) and you delete the local source once you have uploaded to OneDrive (assuming however that the upload was 100% successful, no issues, and your data was not garbled in the middle, or locally or somewhere else ....)

So just to be clear here - the only option for you to clean up your local files & folders is to manually remove them.
